src/components/interaction-designer/flow-editors/import/LanguageAdder.vue

```diff
@@ -0,0 +1,253 @@
+            :valid-state="isValid"
+            @keydown="filterVariant" />
+        </validation-message>
+      </div>
+      <div class="form-group">
+        <label class="form-check-label mt-2 mb-2 mr-2">{{'flow-builder.iso-3166-1-label' | trans}}</label>
+        <vue-multiselect
+          v-model="iso_3166_1"
+          :placeholder="'flow-builder.language-locale-selector-placeholder' | trans"
+          :options="iso_3166_1Locales()"
+          :allow-empty="false"
+          track-by="locale"
+          :custom-label="customLocaleLabel"
+          :searchable="true" />
+      </div>
+      <div class="form-group">
+        <label class="form-check-label mt-2 mb-2 mr-2">{{'flow-builder.bcp-47-label' | trans}}</label>
+        <validation-message
+          #input-control="{ isValid }"
+          message-key="language/new_language/bcp_47">
+          <input
+            v-model="newLanguage.bcp_47"
+            name="bcp_47"
+            type="text"
+            class="form-control"
+            disabled>
+        </validation-message>
+      </div>
+    </b-modal>
+  </div>
+</template>
+
+<script lang="ts">
+
+import {BModal} from 'bootstrap-vue'
+import Lang from '@/lib/filters/lang'
+import Routes from '@/lib/mixins/Routes'
+import Vue from 'vue'
+import {Component} from 'vue-property-decorator'
+import {mixins} from 'vue-class-component'
+import {iso6393} from 'iso-639-3'
+
+import {
+  map,
+  keys,
+  isEmpty,
+} from 'lodash'
+
+import {
+  ILanguage,
+  IContext,
+} from '@floip/flow-runner'
+import {Action, namespace, State} from 'vuex-class'
+
+import {IdGeneratorUuidV4} from '@floip/flow-runner/dist/domain/IdGeneratorUuidV4'
+
+import ValidationMessage from '@/components/common/ValidationMessage.vue'
+import TextEditor from '@/components/common/TextEditor.vue'
+import VueMultiselect from 'vue-multiselect'
+
+const countries = require('i18n-iso-countries')
+countries.registerLocale(require('i18n-iso-countries/langs/en.json'))
+countries.registerLocale(require('i18n-iso-countries/langs/fr.json'))
+
+const importVuexNamespace = namespace('flow/import')
+const validationVuexNamespace = namespace('validation')
+
+@Component({
+  components: {
+    BModal,
+    ValidationMessage,
+    VueMultiselect,
+    TextEditor,
+  },
+})
+class LanguageAdder extends mixins(Lang, Routes) {
+  addLanguageError = ''
+  newLanguage = {
+    id: '',
+    label: '',
+    iso_639_3: '',
+    variant: '',
+    bcp_47: '',
+  }
+  selected_iso_639_3: any = {}
+  selected_iso_3166_1: any = {}
+  iso_639_3Tags: any[] = iso6393
+  // There is no obvious way to narrow down ISO 3166-1 from iso 639 languages:
+  // https://www.rfc-editor.org/rfc/rfc5646.html#section-4.2
+  // As a result, we don't attempt to filter this list depending on the iso 639 selection
+  // We use Alpha-2 Codes as apparently they are "the most widely used"
+  // https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+  resetLanguage(): void {
+    this.newLanguage = {
+      id: '',
```

Comment:
Same as above, can we initiate from ILanguage here ? `this.newLanguage = {} as ILanguage`